DIPs/DIP1xxx-SSS.md

```diff
@@ -0,0 +1,184 @@
+### `__move_post_blt`'s implementation
+
+`__move_post_blt` SHOULD be defined in a manner that is compatible with the following code:
+
+```D
+void __move_post_blt(S)(ref S newLocation, ref S oldLocation) nothrow if( is(S==struct) ) {
+    foreach(memberName; __traits(allMembers, S)) {
+        static if( is( typeof(__traits(getMember, S, memberName))==struct ) ) {
+            mixin("__move_post_blt( newLocation." ~ memberName ~ ", oldLocation." ~ memberName ~ " );");
+        }
+    }
+
+    static if( __traits(hasMember, S, "opPostMove") ) {
+        newLocation.opPostMove(oldLocation);
+    }
+}
+```
+
+Please note that `S` might also be shared, immutable or const.
+
+### `opPostMove`
+
+As should be obvious from the definition of `__move_post_blt`, `opPostMove`, if defined, MUST be a `nothrow` function
+that updates the external/internal references after they have already been copied. Implementors SHOULD also make it `@nogc` and
+either `@safe` or `@trusted`.
+
+Implementors MAY define `opPostMove` for const and/or immutable instances. If they do, the implementation code MAY safely modify
+the data in the destination location for the object, as that has no pointers pointing at it. Such modifications will require a
+cast.
+
+Whether it is safe to modify external data pointed to by pointers stored in the struct (such as the case for an intrusive linked
+list), or the data in the source address of the move, heavily depends on the specifics of the implementation. The user
+documentation for `opPostMove` MUST explain what is guaranteed to be safe and what is not.
+
+Implementors may also choose to not define const/immutable versions of `opPostMove`. This will results in a compile-time error should
+the compiler try to move such a struct.
+
+The documentation for `opPostMove` MUST also emphasize that while manipulating the memory at the `opPostMove` source location is allowed,
+the memory will be effectively freed with no destruction immediately after the function's return. Implementors SHOULD be
+encouraged to define the argument to `opPostMove` to be `const ref` to gain some compiler protection against accidental manipulation.
+This does not harm the implementor's access to the data, as she already has a copy at the destination location.
+
+### Code emitted by the compiler on move
+
+When moving a struct, the compiler should call `__move_post_blt` on the struct, giving it both new and old instances.
+
+### `opPostMove` Decoration Considerations
+
+Ideally, `opPostMove` should be `@nogc`, `nothrow` and either `@safe` or `@trusted`. If that doesn't happen, trying to compile
+code that moves a struct from context that is `nothrow`, `@nogc` or `@safe` would result in a compilation error.
+
+We could force these by decorating `__move_post_blt` itself as `nothrow @nogc @safe`, thus not allowing `opPostMove` to be defined
+any other way (we cannot otherwise force `opPostMove` to be defined any specific way, as it is being defined by the user). I chose
```

Comment:
"(we cannot otherwise force `opPostMove` to be defined any specific way, as it is being defined by the user)" -> ~(we cannot otherwise force `opPostMove` to be defined any specific way, as it is being defined by the user)~